vault/README.md

```diff
@@ -0,0 +1,110 @@
+metadata:
+  name: myapp
+  labels:
+    app: myapp
+  annotations:
+    # NOT HERE!!!
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: myapp
+  template:
+    metadata:
+      labels:
+        app: myapp
+      annotations:
+        # Vault sidecar code goes here, inside the template.
+        vault.hashicorp.com/agent-inject: 'true'
+        vault.hashicorp.com/agent-inject-token: 'true'
+        vault.hashicorp.com/auth-path: auth/k8s-gold  # This was tricky.  Be sure to use k8s-silver, k8s-gold, or k8s-golddr
+        vault.hashicorp.com/namespace: platform-services
+        vault.hashicorp.com/role: abc123-nonprod  # licenseplate-nonprod or licenseplate-prod are your options
+        # We don't really know if this key needs to match the secret or not...please update if you know.
+        vault.hashicorp.com/agent-inject-secret-microservices-secret-dev: abc123-nonprod/microservices-secret-dev
+        vault.hashicorp.com/agent-inject-template-microservices-secret-dev: |
+          {{- with secret "abc123-nonprod/microservices-secret-dev" }}
+          export dev_database_host="{{ .Data.data.dev_database_host }}"
+          export dev_database_name="{{ .Data.data.dev_database_name }}"
+          {{- end `}} }}
+
+        # This is another sample. This set uses some HELM chart variable replacements. There's a bit of magic with the ` symbol in the agent-inject-template section. It also required some additional braces {}. You can use this as an example of what worked for me.
+        vault.hashicorp.com/agent-inject-secret-microservices-secret-debug: {{ .Values.global.licenseplate }}-{{ .Values.global.vault_engine }}/microservices-secret-debug
+        vault.hashicorp.com/agent-inject-template-microservices-secret-debug: |
+          {{`{{- with secret `}}"{{ .Values.global.licenseplate }}-{{ .Values.global.vault_engine }}/microservices-secret-debug"{{` }}
+          export dev_hostname="{{ .Data.data.dev_hostname }}"
+          export dev_toolbox="{{ .Data.data.dev_toolbox }}"
+          {{- end `}} }}
+    spec:
+      serviceAccount: abc123-vault
+      # or for HELM with var replacements use:
+      #serviceAccount: {{ .Values.global.licenseplate }}-vault
+
+  ```
+
+Note the additional line under spec that I added. This is the service account that is needed to connect to the Vault. This account has been created already for you in Openshift so on the surface it's straight forward. You may notice another one that's used in the demo:
+```        serviceAccountName: LICENSE-vault```
+I believe the serviceAccountName is used for Openshift and ServiceAccount is used by Kubernetes. I've yet to determine the differences or why there is a difference, but in any case, using ServiceAccount did the trick for me while using Deployments. Having both did not hurt. This is more the Kubernetes way of doing things than the Openshift method which I've seen uses DeploymentConfigs and possibly that's where serviceAccountName comes into play?
```

Comment:
```suggestion
Note the additional line under spec that I added. This is the service account that is needed to connect to the Vault. This account has been created already for you in Openshift so on the surface it's straight forward. You may notice another one that's used in the demo: `serviceAccountName: LICENSE-vault`
```

According to `oc explain pod.spec.serviceAccount` that one is deprecated and `serviceAccountName` is the new/correct one to use.

Also improved the markdown a bit.